stackrox/stackrox
# ROX-12186: Add Type column to image integration tables for Clairify, Google, Quay

## Description

Display **Type** column for image integrations which can have either or both of two categories.

### Changed files

1. Edit src/Containers/Integrations/IntegrationForm/Forms/ClairifyIntegrationForm.tsx
    * Replace module level type definition with `ClarifyImageIntegration` type.
    * Correct default value of `numConcurrentScans` property.
    * Delete obsolete `clusters` property.

2. Edit src/Containers/Integrations/IntegrationForm/Forms/GoogleIntegrationForm.tsx
    * Replace module level type definition with `GoogleImageIntegration` type.
    * Delete obsolete `clusters` property.

3. Edit src/Containers/Integrations/IntegrationForm/Forms/QuayIntegrationForm.tsx
    * Replace module level type definition with `QuayImageIntegration` type.
    * Delete obsolete `clusters` property.

4. Edit src/Containers/Integrations/utils/integrationUtils.ts
    * Replace `ImageIntegrationCategory[]` with `(Category0 | Category1)[]` as Dave recommended in https://github.com/stackrox/stackrox/pull/2656/files#r943833228

5. Edit src/Containers/Integrations/utils/tableColumnDescriptor.ts
    * Add column descriptors with type assertion in argument of `getCategoriesTextForClairifyScanner` function

6. Edit src/types/imageIntegration.proto.ts
    * Replace incorrect underscore case `skip_test_integration` with correct camel case `skipTestIntegration` for property name in `BaseImageIntegration` type
    * Specify categories for image integrations: Clairify, Google, Quay

### Residue

1. Replace more type definitions related to integrations with import from files in types folder.

## Checklist

## Testing Performed

### manual testing without feature flag

1. `yarn deploy-local` in ui
2. `yarn start` in ui

    * Visit /main/integrations/imageIntegrations/clairify: See table does not have a **Type** column

### manual testing with feature flag

1. `export ROX_QUAY_ROBOT_ACCOUNTS=true` in ui
2. `yarn deploy-local` in ui
3. `yarn start` in ui

    * Visit /main/integrations/imageIntegrations/clairify: See table does have a **Type** column
        ![clairify-table-type](https://user-images.githubusercontent.com/11862657/184638250-5011d1ee-d8bb-4110-9974-fd2d5b2f65fb.png)

    * Visit /main/integrations/imageIntegrations/quay: See table does have a **Type** column
        ![table-quay-type](https://user-images.githubusercontent.com/11862657/184638267-b2ebf8f5-5124-436a-a2d0-2e6c5067b0ce.png)

### integration testing without feature flag

1. `yarn cypress-open` in ui/apps/platform
    * integrations/imageIntegration

### integration testing with feature flag

1. `export CYPRESS_ROX_QUAY_ROBOT_ACCOUNTS=true` in ui
2. `yarn cypress-open` in ui/apps/platform
    * integrations/imageIntegration
